python/examples/volume/org.xen.xapi.storage.simple-file/plugin.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify 
+# it under the terms of the GNU Lesser General Public License as published 
+# by the Free Software Foundation; version 2.1 only.
+#
+# This program is distributed in the hope that it will be useful, 
+# but WITHOUT ANY WARRANTY; without even the implied warranty of 
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+import os
+import sys
+
+import xapi.storage.api.v5.plugin
+from xapi.storage import log
+
+
+class Implementation(xapi.storage.api.v5.plugin.Plugin_skeleton):
+
+    def diagnostics(self, dbg):
+        return "No diagnostic data available"
+
+    def query(self, dbg):
+
+        config = {'path': 'Folder path for SR'}
+
+        return {
+            "plugin": "simple-file",
+            "name": "Simple example file SR",
+            "description": ("This is an example SR, it creates files in a directory"),
+            "vendor": "Citrix",
+            "copyright": "(C) 2019 Citrix Inc",
+            "version": "3.0",
+            "required_api_version": "5.0",
+            "features": [
+                "SR_ATTACH",
+                "SR_DETACH",
+                "SR_CREATE",
+                "VDI_CREATE",
+                "VDI_DESTROY",
+                "VDI_ATTACH",
+                "VDI_DETACH",
+                "VDI_ACTIVATE",
+                "VDI_DEACTIVATE",
+                "VDI_UPDATE",
+                "VDI_RESIZE",
+                "SR_METADATA",
```

Comment:
I'm not sure about the exact meaning of this, but it looks like this is needed only if the SR supports disaster recovery. SInce this example SR doesn't even support set_name or set_description I think we could safely remove this particular feature.

Reply:
Yes, very true.